src/function.py

```diff
@@ -264,7 +264,7 @@ def __jump_to_next_loc(self, block: tac_cfg.TACBasicBlock,
             # We can discard blocks in the body of the function being identified
             # since we want to discover new blocks, not old ones
             if not in_func and block not in self.invoc_pairs.keys() and \
-                block.ident() in str(exit_stack) and block not in body:
```

Comment:
Hm actually I think the reason for this, on second thought, is in order to deal with blocks that have been duplicated during dataflow analysis but not yet merged.

It compares strings because duplicate blocks are differentiated by the information in their ident suffixes.

Reply:
But I agree this str(exit_stack) thing is pretty horrible.

Reply:
Yeah, we need a better way to identify duplicated blocks than just the ident suffix O_o